services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/attachment/application/ApplicationAttachmentMigrator.kt

```diff
@@ -0,0 +1,109 @@
+package fi.hel.haitaton.hanke.attachment.application
+
+import fi.hel.haitaton.hanke.attachment.application.ApplicationAttachmentContentService.Companion.generateBlobPath
+import fi.hel.haitaton.hanke.attachment.azure.Container
+import fi.hel.haitaton.hanke.attachment.common.ApplicationAttachmentContentEntity
+import fi.hel.haitaton.hanke.attachment.common.ApplicationAttachmentContentRepository
+import fi.hel.haitaton.hanke.attachment.common.ApplicationAttachmentEntity
+import fi.hel.haitaton.hanke.attachment.common.ApplicationAttachmentRepository
+import fi.hel.haitaton.hanke.attachment.common.AttachmentContent
+import fi.hel.haitaton.hanke.attachment.common.AttachmentContentNotFoundException
+import fi.hel.haitaton.hanke.attachment.common.AttachmentNotFoundException
+import fi.hel.haitaton.hanke.attachment.common.AttachmentValidator
+import fi.hel.haitaton.hanke.attachment.common.FileClient
+import java.util.UUID
+import mu.KotlinLogging
+import org.springframework.data.repository.findByIdOrNull
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+
+private val logger = KotlinLogging.logger {}
+
+@Service
+class ApplicationAttachmentMigrator(
+    private val fileClient: FileClient,
+    private val attachmentRepository: ApplicationAttachmentRepository,
+    private val contentRepository: ApplicationAttachmentContentRepository,
+) {
+    /** Find an attachment that is not yet migrated, i.e. still has content in the db. */
+    @Transactional(readOnly = true)
+    fun findAttachmentWithDatabaseContent(): ApplicationAttachmentWithContent? {
+        return attachmentAmountToMigrate().let {
+            if (it == 0L) {
+                return null
+            }
+            logger.info {
+                logger.info { "There are un-migrated application attachments left, count = $it" }
+            }
+            pickForMigration().let { (attachmentEntity, contentEntity) ->
+                ApplicationAttachmentWithContent(
+                    id = attachmentEntity.id!!,
+                    applicationId = attachmentEntity.applicationId,
+                    content =
+                        AttachmentContent(
+                            fileName = attachmentEntity.fileName,
+                            contentType = attachmentEntity.contentType,
+                            bytes = contentEntity.content,
+                        ),
+                )
+            }
+        }
```

Comment:
There's some unnecessary nesting here.
```suggestion
        val amount = attachmentAmountToMigrate()
        if (amount == 0L) {
            return null
        }
        logger.info {
            logger.info { "There are un-migrated application attachments left, count = $amount" }
        }
        val (attachmentEntity, contentEntity) = pickForMigration()
        return ApplicationAttachmentWithContent(
            id = attachmentEntity.id!!,
            applicationId = attachmentEntity.applicationId,
            content =
                AttachmentContent(
                    fileName = attachmentEntity.fileName,
                    contentType = attachmentEntity.contentType,
                    bytes = contentEntity.content,
                ),
        )
```

Reply:
Fixed as suggested.